control/passivity.py

```diff
@@ -174,8 +170,18 @@ def P_pos_def_constraint(n):
 
     # crunch feasibility solution
     cvx.solvers.options['show_progress'] = False
-    sol = cvx.solvers.sdp(c, Gs=Gs, hs=hs)
-    return sol["x"]
+    try:
+        sol = cvx.solvers.sdp(c, Gs=Gs, hs=hs)
+        return sol["x"]
+
+    except ZeroDivisionError as e:
+        print(e)
+        print(
+            """The system is probably ill conditioned.
+            Consider perturbing the system matrices a small amount."""
+        )
+        raise(ZeroDivisionError)
```

Comment:
1. On a UNIX system this outputs the error message to stdout but the Exception itself to stderr.
2. The multiline string is not dedented automatically, so it produces weirdly formatted output:

```python
>>> print(
...             """The system is probably ill conditioned.
...             Consider perturbing the system matrices a small amount."""
...         )
The system is probably ill conditioned.
            Consider perturbing the system matrices a small amount.
```

```suggestion
        raise ValueError("The system is probably ill conditioned. "
                         "Consider perturbing the system matrices by a small amount."
        ) from e
```

```python
>>> try:
...     1/0
... except ZeroDivisionError as e:
...         raise ValueError("The system is probably ill conditioned. "
...                          "Consider perturbing the system matrices by a small amount."
...         ) from e
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
ValueError: The system is probably ill conditioned. Consider perturbing the system matrices by a small amount.

```